### DISCUSSION TITLE
Can't set Default Value for Relationship Picker

### DISCUSSION THREAD

**Discussion by pbarmak** [Budibase/budibase#7687] [Help] [answered]
I have a form that uses a Relationship Picker (off of a relationship between two MySQL tables). I tried setting the Default value to a known value that would show up in the picker (the number 10 in my case). But when I go to preview the app and run the form, the picker just shows "null" instead of selecting the default.

How can I set the default value of a relationship picker field?

This is on version 1.3.11, if that helps.

**Comment by melohagan** [comment, marked as answer]
Hey @pbarmak 

You'll probably need to use the encoded value, for example: `%5B1%5D`

![Screenshot 2022-09-13 at 15 41 23](https://user-images.githubusercontent.com/101575380/189931229-c6a5fdf4-f83e-421b-94ff-f8643f5fb674.png)
![Screenshot 2022-09-13 at 15 41 37](https://user-images.githubusercontent.com/101575380/189931301-6e203ba2-8523-4b2b-ab55-23384a74e1d9.png)

To verify, you can click show *Auto columns* and copy the autogenerated **ID**
![Screenshot 2022-09-13 at 15 43 37](https://user-images.githubusercontent.com/101575380/189931822-a6b5da1b-ef39-4d50-9e55-1a8c7f68f651.png)

**Reply by pbarmak** [reply]
Yeah, that does work. But, is this not a bug? It feels like it should handle just a straight value (or a binding of a field value without having to put those special chars in).

**Reply by melohagan** [reply]
I had raised this issue before: https://github.com/Budibase/budibase/issues/6194

I closed it because it was stale, but I'll reopen it, because as you rightly point out it's not ideal!

**Reply by pbarmak** [reply]
Thank you!

**Reply by NicolasFabre81** [reply]
Hello @melohagan  , what about this behaviour in current V2.4 release ? I don't manage to make it work, neither with the normal id ("4" for me) nor with the encoded one "%5B4%5D".
It gives me :
<img width="267" alt="image" src="https://user-images.githubusercontent.com/28592069/227583466-1a7d3b27-8cd6-4b99-882c-3a419fabb5a9.png">

**Reply by melohagan** [reply]
Hey @NicolasFabre81 Would you be able to try the JavaScript binding: `return 4` incase it's a typing issue?

**Reply by NicolasFabre81** [reply]
I just found the workaround : in the relationship field default value I put :
`return helpers.encodeURI('[') +  helpers.encodeURI($("URL.MyID")) + helpers.encodeURI(']')`
